sdk/attestation/azure-security-attestation/azure/security/attestation/_models.py

```diff
@@ -69,242 +79,244 @@ class PolicyCertificatesModificationResult(object):
+        # type: (Dict[str,Any]) -> None
+        self._issuer = kwargs.pop("issuer")  # type: Union[str, None]
+        self._unique_identifier = kwargs.pop(
+            "unique_identifier", None
+        )  # type: Union[str, None]
+        self._nonce = kwargs.pop("nonce", None)  # type: Union[str, None]
+        self._version = kwargs.pop("version")  # type: str
+        self._runtime_claims = kwargs.pop(
+            "runtime_claims", None
+        )  # type: Union[Dict, None]
+        self._inittime_claims = kwargs.pop(
+            "inittime_claims", None
+        )  # type: Union[Dict, None]
+        self._policy_claims = kwargs.pop(
+            "policy_claims", None
+        )  # type: Union[Dict, None]
+        self._verifier_type = kwargs.pop("verifier_type")  # type: str
+        self._policy_signer = kwargs.pop(
+            "policy_signer", None
+        )  # type: Union[AttestationSigner, None]
+        self._policy_hash = kwargs.pop("policy_hash")  # type: str
+        self._is_debuggable = kwargs.pop("is_debuggable")  # type: bool
+        self._product_id = kwargs.pop("product_id")  # type: int
+        self._mr_enclave = kwargs.pop("mr_enclave")  # type: str
+        self._mr_signer = kwargs.pop("mr_signer")  # type: str
+        self._svn = kwargs.pop("svn")  # type: int
+        self._enclave_held_data = kwargs.pop(
+            "enclave_held_data", None
+        )  # type: Union[bytes, None]
+        self._sgx_collateral = kwargs.pop("sgx_collateral")  # type: Dict
 
     @classmethod
     def _from_generated(cls, generated):
-        #type:(GeneratedAttestationResult) -> AttestationResult
+        # type: (GeneratedAttestationResult) -> AttestationResult
+        if not generated:
+            return None
         return AttestationResult(
             issuer=generated.iss,
-            confirmation=generated.cnf,
             unique_identifier=generated.jti,
             nonce=generated.nonce,
             version=generated.version,
             runtime_claims=generated.runtime_claims,
             inittime_claims=generated.inittime_claims,
             policy_claims=generated.policy_claims,
             verifier_type=generated.verifier_type,
-            policy_signer=AttestationSigner._from_generated(generated.policy_signer) if generated.policy_signer else None,
+            policy_signer=AttestationSigner._from_generated(  # pylint: disable=protected-access
+                generated.policy_signer
+            ),
             policy_hash=generated.policy_hash,
             is_debuggable=generated.is_debuggable,
             product_id=generated.product_id,
             mr_enclave=generated.mr_enclave,
             mr_signer=generated.mr_signer,
             svn=generated.svn,
             enclave_held_data=generated.enclave_held_data,
-            sgx_collateral=generated.sgx_collateral)
+            sgx_collateral=generated.sgx_collateral,
+        )
 
     @property
```

Comment:
Based on [this](https://github.com/Azure/azure-sdk/blob/main/docs/python/_includes/example_client.py#L26), my understanding was that attributes are preferred over properties. But I could be wrong! @annatisch, is there a real preference, and/or have I missed some context leading to this class structure?

Reply:
I chose properties over attributes because it allowed me significantly more room to document the meaning of the various fields since the interactions between them are non-trivial.

In addition, some of the properties required calculations (`policy_signer`) which are expensive. Making them properties made it a pay-to-play cost instead of a constant overhead.

Reply:
It seems like properties are fine, especially in this case. I'm also fond of how much easier they are to document, so I agree that they're a good choice here. Thanks for clearing up the architects' stance on this!